app/src/modules/shell/shell.c

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ #include <stdlib.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/shell/shell_uart.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/zbus/zbus.h>
+#include <date_time.h>
+#include <zephyr/task_wdt/task_wdt.h>
+#include <modem/nrf_modem_lib_trace.h>
+
+#include "message_channel.h"
+
+LOG_MODULE_REGISTER(shell, CONFIG_APP_SHELL_LOG_LEVEL);
+
+/* Register subscriber */
+ZBUS_SUBSCRIBER_DEFINE(shell, CONFIG_APP_SHELL_MESSAGE_QUEUE_SIZE);
+
+enum zbus_test_type {
+	PING,
+};
+
+ZBUS_CHAN_DECLARE(ZBUS_TEST_CHAN);
+ZBUS_CHAN_DEFINE(ZBUS_TEST_CHAN,
+		 enum zbus_test_type,
+		 NULL,
+		 NULL,
+		 ZBUS_OBSERVERS(shell),
+		 ZBUS_MSG_INIT(0)
+);
+
+#define UART_DEVICE_NODE DT_CHOSEN(zephyr_shell_uart)
+static const struct device *const shell_uart_dev = DEVICE_DT_GET(UART_DEVICE_NODE);
+
+static void uart_disable_handler(struct k_work *work)
+{
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_UART
+	int err = nrf_modem_lib_trace_level_set(NRF_MODEM_LIB_TRACE_LEVEL_OFF);
+	if (err) {
+		LOG_ERR("nrf_modem_lib_trace_level_set() failed with err = %d.", err);
+	}
+#endif
+
+	if (device_is_ready(shell_uart_dev)) {
+		pm_device_action_run(shell_uart_dev, PM_DEVICE_ACTION_SUSPEND);
+	}
+}
+
+static void uart_enable_handler(struct k_work *work)
+{
+	if (device_is_ready(shell_uart_dev)) {
+		pm_device_action_run(shell_uart_dev, PM_DEVICE_ACTION_RESUME);
+	}
+
+#ifdef CONFIG_NRF_MODEM_LIB_TRACE_BACKEND_UART
+	int err = nrf_modem_lib_trace_level_set(NRF_MODEM_LIB_TRACE_LEVEL_FULL);
+	if (err) {
+		LOG_ERR("nrf_modem_lib_trace_level_set() failed with err = %d.", err);
+	}
+#endif
+
+	LOG_INF("UARTs enabled\n");
+}
+
+static K_WORK_DEFINE(uart_disable_work, &uart_disable_handler);
+static K_WORK_DELAYABLE_DEFINE(uart_enable_work, &uart_enable_handler);
+
+void uart_disable(void)
+{
+	LOG_INF("Disabling UARTs requested");
+	k_work_submit(&uart_disable_work);
+}
+
+void uart_enable(void)
+{
+	LOG_INF("Enabling UARTs requested");
+	k_work_schedule(&uart_enable_work, K_NO_WAIT);
+}
+
+static int cmd_uart_disable(const struct shell *sh, size_t argc,
+                         char **argv)
+{
+	int sleep_time;
+
+	if (argc != 2) {
+		LOG_ERR("disable: invalid number of arguments");
+		return -EINVAL;
+	}
+
+	sleep_time = atoi(argv[1]);
+	if (sleep_time < 0) {
+		LOG_ERR("disable: invalid sleep time");
+		return -EINVAL;
+	}
+
+	if (sleep_time > 0) {
+		LOG_INF("disable: disabling UARTs for %d seconds", sleep_time);
+	} else {
+		LOG_INF("disable: disabling UARTs indefinitely");
+	}
+	k_sleep(K_MSEC(500)); /* allow little time for printing the notification */
+	k_work_submit(&uart_disable_work);
```

Comment:
You can use `uart_disable()` instead here, or remove that function as it's currently unused.
If you decide to use it, it should be made static. Same for `uart_enable()`

Reply:
I'll just remove those 2 functions